### PR TITLE
Usability improvements for generate_incremental_table

### PR DESCRIPTION
**Pull request by jklukas**

### CODE REVIEW

File: script/generate_incremental_table [hunk 1]
```diff
@@ -78,8 +78,8 @@ def query(day, entrypoint, destination_table, query_options, query_file):
             "--replace",
             "--destination_table=" + destination_table + day.strftime("$%Y%m%d"),
             "--parameter=submission_date:DATE:" + day.isoformat(),
-            "--time_partitioning_field=submission_date",
```
Comment (jklukas) [inline script/generate_incremental_table]: If an existing table is clustered, either neither or both of --time_partitioning_field and --clustering_fields must be specified, or neither.

File: script/generate_incremental_table [hunk 2]
```diff
@@ -101,7 +101,9 @@ def query(day, entrypoint, destination_table, query_options, query_file):
 
 
 def main():
-    args = parser.parse_args()
+    args, unknown_args = parser.parse_known_args()
```
Comment (jklukas) [inline script/generate_incremental_table]: This allows the user to pass `--dataset`, etc.

File: script/generate_incremental_table [hunk 3]
```diff
@@ -78,8 +79,8 @@ def query(day, entrypoint, destination_table, query_options, query_file):
             "--replace",
             "--destination_table=" + destination_table + day.strftime("$%Y%m%d"),
```
Comment (relud) [inline script/generate_incremental_table]: While we're at it, we should set a default `dataset_id`
```suggestion
             "--dataset_id=telemetry",
             "--destination_table=" + destination_table + day.strftime("$%Y%m%d"),
```

File: script/generate_incremental_table (hunk 2 above)
Comment (relud) [reply script/generate_incremental_table]: they would be included in `query_options` if preceded with a `--` option, but I love that this takes away the restriction on where they are placed. 🎉

**Pull request merged**